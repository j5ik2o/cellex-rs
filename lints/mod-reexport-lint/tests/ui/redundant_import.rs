mod hello {
  pub mod actor_process {
    pub struct ActorProcess;
  }
}

use crate::hello::actor_process::ActorProcess;

fn main() {}
