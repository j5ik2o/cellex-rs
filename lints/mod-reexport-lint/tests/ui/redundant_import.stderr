warning: `crate::hello::actor_process::ActorProcess` のような冗長な import です
  --> $DIR/redundant_import.rs:7:34
   |
LL | use crate::hello::actor_process::ActorProcess;
   |                                  ^^^^^^^^^^^^
   |
   = help: `mod actor_process;` と `pub use actor_process::ActorProcess;` をこのファイルで宣言して再エクスポートしてください
   = note: AI向けアドバイス: 1. 同名モジュールを `mod` 宣言で読み込み 2. 公開が必要な型を `pub use` でまとめて再エクスポート 3. 呼び出し側の `use` を削除して親モジュールから直接利用しましょう。
   = note: `#[warn(redundant_mod_import)]` on by default

warning: 1 warning emitted

